public/app/features/explore/state/actions.ts

```diff
@@ -706,24 +706,31 @@ export function splitOpen(options?: { datasourceUid: string; query: string }): T
     const queryState = getState().location.query[ExploreId.left] as string;
     const urlState = parseUrlState(queryState);
 
-    // TODO: Instead of splitting and then setting query/datasource we may probably do it in one action call
-    rightState.queries = leftState.queries.slice();
-    rightState.urlState = urlState;
-    dispatch(splitOpenAction({ itemState: rightState }));
-
     if (options) {
-      // TODO: This is hardcoded for Jaeger right now. Need to be changed so that target datasource can define the
-      //  query shape.
+      rightState.queries = [];
+      rightState.graphResult = undefined;
+      rightState.logsResult = undefined;
+      rightState.tableResult = undefined;
+      rightState.queryKeys = [];
+      urlState.queries = [];
+      rightState.urlState = urlState;
+
+      dispatch(splitOpenAction({ itemState: rightState }));
```

Comment:
This worked before but triggered few cycles of setting and resetting queries resulting in blinking of different query and sometimes race conditions.